astro/src/content/docs/get-started/start-here/step-3.mdx

```diff
@@ -0,0 +1,84 @@
+---
+title: "Step 3 - Protected pages"
+description: Learn how to use FusionAuth by using an example application that has a protected page defined.
+section: get started
+subcategory: start here
+navOrder: 0
+codeRoot: https://raw.githubusercontent.com/fusionauth/fusionauth-example-get-started/main
+---
+
+<div className="border border-green-700 px-3 rounded-md">
+  <i className="fa-duotone fa-stopwatch fa-2x mr-2 text-green-700 align-middle"/> Estimated time to complete this step: **5 minutes**
+</div>
+
+## Protecting pages
+
+Now that we have the FusionAuth and the example app running, let's take a look at how the application protects pages that require a user to be logged in.
+
+This process is also know as access control. It is the process of controlling the users that have access to specific pages and functions of an application. The first step is to protect a page such that only users that have logged in can view it.
+
+Open the file `src/index.mts` and find the route named `/account`. It should look something like this:
+
+```typescript
+app.get("/account", async (req, res) => {
+  if (!await sdk.userHasAccess(req, res, ["admin", "user"])) {
+    res.redirect(302, '/login');
+    return;
+  }
+
+  res.sendFile(path.join(__dirname, '../templates/account.html'));
+});
+```
+
+The first check is where the application determines if the user is logged in or not. This is done by calling the SDK function `userHasAccess` (you can ignore the last parameter, which we will cover in the next section). This function returns true if the user has access, which implicitly verifies that they are logged in. Let's take a look at this function.
+
+Open the file `src/sdk.ts` and find the function named `userHasAccess`. One note is that this application uses `jose` which is a Node library used for validating and parsing [JSON Web Tokens (JWTs)](/docs/lifecycle/authenticate-users/login-api/json-web-tokens). If you want to learn more about `jose`, you can visit the project at Github here: https://github.com/panva/jose
+
+The `userHasAccess` method begins by loading the user from the request. This is accomplished by calling the `getUser` function of the SDK, which returns the user's access token. If the access token comes back `null`, that means the user is not logged in and this function returns false.
+
+```typescript
+async userHasAccess(req: Request, res: Response, roles: Array<string>): Promise<boolean> {
+  const jwt = await this.getUser(req, res);
+
+  // ...
+}
+```
+
+Let's take a quick peek at the `getUser` function of the SDK. This function looks like this:
+
+```typescript
```

Comment:
I would prefer if we used RemoteCode to pull in these fragments (like here: https://github.com/FusionAuth/fusionauth-site/blob/main/astro/src/content/quickstarts/quickstart-ruby-rails-web.mdx?plain=1#L72 ) because it means the code will always be in sync with the example application repo. You can also use tags if you want to pull in fragments and you can also pull in a specific value from a JSON file including a kickstart file (more doco here: https://github.com/FusionAuth/fusionauth-astro-components )